### PR TITLE
Added residual minimization for SPQE and UCCNPQE

### PR DESCRIPTION
**Pull request by imagoulas**
## Description
Added residual minimization functionality for UCCNPQE and SPQE classes. The minimization is based on scipy.

Added a couple of tests for the new features.

In a future PR, the scipy solver for UCCNVQE and ADAPTVQE will be replaced by the scipy solver in the optimizer.py file.

## User Notes
- [x ] Features added
- [ ] Changes to compilation (if any)

## Checklist
- [x ] Added/updated tests of new features
- [ ] Removed comments in input files
- [x ] Documented source code
- [ ] Checked for redundant headers/imports
- [ ] Checked for consistency in the formatting of the output file
- [x ] Ready to go!

### CODE REVIEW

File: src/qforte/ucc/uccnpqe.py [hunk 1]
```diff
@@ -267,6 +275,13 @@ def get_residual_vector(self, trial_amps):
 
         return residuals
 
+    def get_sum_residual_square(self, tamps):
```
Comment (JonathonMisiewicz) [inline src/qforte/ucc/uccnpqe.py:278]: Does it make more sense to have this as part of `uccpqeabc.py`?

File: src/qforte/ucc/uccnpqe.py [hunk 2]
```diff
@@ -167,7 +170,12 @@ def print_summary_banner(self):
         print('Number of non-zero res element evaluations:  ', int(self._res_vec_evals)*self._n_nonzero_params)
 
     def solve(self):
```
Comment (JonathonMisiewicz) [inline src/qforte/ucc/uccnpqe.py:172]: Again, I wonder if this belongs in `uccpqeabc.py`.

File: src/qforte/ucc/uccnpqe.py (hunk 1 above)
Comment (imagoulas) [reply src/qforte/ucc/uccnpqe.py:278]: I agree with you. In fact, I wanted to do that but I forgot.

File: src/qforte/ucc/uccnpqe.py (hunk 2 above)
Comment (imagoulas) [reply src/qforte/ucc/uccnpqe.py:172]: I agree again. This one did not occur to me. I will do the transfer of code and push.

**Comment by JonathonMisiewicz** [comment]
Please add the changes to `spqe.py` as well.

**Review by JonathonMisiewicz — APPROVED** [review]

**Pull request merged**